pkg/process/process_manager.go

```
@@ -111,6 +127,60 @@ func (pm *Manager) Shutdown() {
 	close(pm.shutdownCh)
 }
 
+func (pm *Manager) checkMountPointStatusForEngine() {
+	var processToUpdate []*Process
+	volumeMountPointMap, err := getVolumeMountPointMap()
+	if err != nil {
+		logrus.Errorf("Failed to get all volume mount points\n")
+	}
+
+	pm.lock.RLock()
+	for _, p := range pm.processes {
+		p.lock.Lock()
+		if isEngineProcess(p) && p.State == StateRunning {
+
+			nameSlices := strings.Split(p.Name, "-")
+			volumeNameSHA := sha256.Sum256([]byte(strings.Join(nameSlices[:len(nameSlices)-2], "-")))
+
+			if mp, exists := volumeMountPointMap[hex.EncodeToString(volumeNameSHA[:])]; exists {
+				if util.IsMountPointReadOnly(mp) {
+					p.Conditions[types.EngineConditionFilesystemReadOnly] = true
+					processToUpdate = append(processToUpdate, p)
+				}
+			}
+		}
+		p.lock.Unlock()
+	}
+	pm.lock.RUnlock()
+
+	for _, p := range processToUpdate {
+		p.UpdateCh <- p
+	}
+
+}
+
+func isEngineProcess(p *Process) bool {
+	return p.PortCount == DefaultEnginePortCount
+}
+
+func getVolumeMountPointMap() (map[string]mount.MountPoint, error) {
+	volumeMountPointMap := make(map[string]mount.MountPoint)
+	mounter := mount.New("")
+	mountPoints, _ := mounter.List()
```

Suggest not to swallow the error.

---

Fixed, thanks!